modules/battery_manager/test/TestPlan.md

```diff
@@ -89,15 +88,13 @@ Test Battery.
 
 **Testing edit (activate/deactivate/add) functionality**
   1. Check that you cannot edit an entry without filling out the required fields: `Instrument`, `Minimum age (days)`, `Maximum age (days)`, `Stage`.
-  2. Check that you can only enter a site that exists.
```

Comment:
@laemtl just for fun. open the inspector and change the value of the dropdown manually to a site that does not exist, try submitting the form and see what happens.

Having a dropdown does not guarantee that only these values can be submitted, it only offers these values but a user can edit the html to whatever they want and still submit it.

Reply:
So maybe clarifying the Test plan with this procedure will help?

Reply:
@ridz1208  I may be wrong here, but just because you change the html doesn't mean that you've triggered a state change to update the actual object that will be submitted. I'm sure there's a way to manipulate the state from the front end if you're clever, but if the center really doesn't exist, the backend will throw a foreign key error. So I'm not sure if this is an issue.

Reply:
I just checked and if you force-submit a centerId that doesn't exist, at the database access stage it sets the centerId to `null` without throwing an error (atleast I can't find any). I think the select drop down is sufficient enough front-end validation, so if you want an error to be thrown in this case, I think it should be handled in the back end. This validation should, in my opinion, be applied to a center/centerId object that can be used throughout LORIS.

Reply:
I think this test case was added exactly for that reason, making sure the backend validates objects before persisting them even if they make it past the front end validations (which can be manipulated by a malicious user)

Reply:
I think a bad request error should be thrown in that specific case